### PR TITLE
softlist.cpp: Add support for a 'notes' field to store information ab…

### PR DESCRIPTION
**Pull request by wilbertpol**
…out a software list or software list item.  [Wilbert Pol]

There is quite a lot of information regarding software for a system, why/how a software list item is broken, or how to use a software list item in the current software lists. Yet this information is only available in xml comments.

By moving this information into a xml tag the information becomes available for a frontend/ui and could be presented to the user.
Having the information in a tag will also allow us to change from xml to a different format without losing any information if we ever want to do so.

As an example I changed the wswan software list as part of this PR.

### CODE REVIEW

**Comment by wilbertpol** [comment]
I see this broke minimaws. I will have to fix that

File: hash/wswan.xml [hunk 1]
```diff
@@ -2212,15 +2209,15 @@ The GIZA chip used on a lot of cartridges is a sram voltage switch.
...
-	-->
 	<software name="robworks" supported="partial">
 		<description>Wonder Borg Robot Works</description>
 		<year>2000</year>
 		<publisher>Bandai</publisher>
+		<notes><![CDATA[
+The Robot Works cartridge has an integrated infrared port for programming an
+external robot (the Wonder Borg).
+The Wonder Borg itself uses an Elan EM78P451AQ MCU (undumped).
+]]></notes>
```
Comment (angelosa) [inline hash/wswan.xml:2220]: Not sure if I like that multi-line breaks indentation wrt the rest of the XML entity when opening in raw form. 
But if the general consensus is to go like this then I won't complain (would mean that front-ends should apply an `lstrip()` for each line, which is trivial to do but hey).

Comment (wilbertpol) [reply hash/wswan.xml:2220]: Yeah, that is a bit of a downside with this. Also in json format it would look similarly ugly.

**Comment by angelosa** [comment]
Any good reason to not apply this? I have plans to handle XMLs via outside frontend, and handling notes inside the XML entities sounds a better option than line commentary.

**Comment by cuavas** [comment]
This is going to hurt query performance for minimaws pretty badly if it stays in this form.  Putting the notes field in the software list table itself is poor design – it will bloat a table that needs to support fast listing queries.  You definitely don’t want to be returning the contents of the notes field in listing queries, either.

Also, have you considered the performance impact of loading and keeping this in memory for big software lists in MAME?
* There’s already a noticeable delay on selecting an Apple II driver while it loads the items from the software lists to show the software selection menu.
* There’s an even bigger delay on trying to start an Apple II driver when it loads the software lists multiple times as it does the validity checks for all the drivers defined in the same source file as the machine you’re starting.  There’s over a second with nothing on the screen after the system/software menu selection window disappears before the window for the emulation session appears.

It’s already bad enough, but accumulating text for the notes elements and storing it with the software list items is going to exacerbate this badly.  The current approach of loading software list information directly from XML files isn’t scaling.

I realise keeping everything in XML comments isn’t a great approach, but I have a bad feeling that this notes field is going to turn into a bucket for all kinds of irrelevant trivia that would be better suited to some wiki (e.g. like SegaRetro for Sega games).  It’s not even entirely clear on whether it’s supposed to be for notes about the software, or notes about MAME’s support for the software.

This feature can be considered on probation:
* The performance impact on minimaws need to be fixed, or it needs to be changed to just ignore notes elements.
* Something has to be done to mitigate the performance impact this will cause on starting drivers, or the software list code needs to be changed to ignore the contents of the notes fields.  We can’t afford to make it any worse than it already is.
* Submissions need to be policed to ensure the notes elements don’t fill up with irrelevant trivia.  If that doesn’t happen, the field needs to be removed.

**Comment by FoxhackDN** [comment]
I mentioned this on Discord, but I don't agree with moving -all- the information from comments to notes fields. Some specific information that is meant to be shown to the user is fine - for example, games and systems that require specific type in commands to start.

But all the comments... that feels a bit excessive. A normal user doesn't need to know that X game might need to be redumped, for instance.

**Comment by cuavas** [comment]
>Some specific information that is meant to be shown to the user is fine - for example, games and systems that require specific type in commands to start.

That’s what `<info name="usage"…` is for – that doesn’t belong in the new notes element.

>But all the comments... that feels a bit excessive. A normal user doesn't need to know that X game might need to be redumped, for instance. (I mean via a note on the software list, since we already have the "game is bad dump / needs redump" tag.)

Yeah sure, we have metadata fields for all sorts of things.  Hence my concern that this is going to become a dumping ground for irrelevant trivia.

**Comment by FoxhackDN** [comment]
> That’s what `<info name="usage"…` is for – that doesn’t belong in the new notes element.

Oh neat, I was not aware of this! I haven't done much with software lists past the few games I've added.

> Yeah sure, we have metadata fields for all sorts of things. Hence my concern that this is going to become a dumping ground for irrelevant trivia.

I agree. That's what mameinfo.dat was for. ;P

**Pull request merged**